delivery_multi_destination/tests/test_delivery_multi_destination.py

```diff
@@ -0,0 +1,168 @@
+        )
+        cls.partner_1 = cls.env["res.partner"].create(
+            {
+                "name": "Test partner 1",
+                "country_id": cls.country_1.id,
+                "property_product_pricelist": cls.pricelist.id,
+            }
+        )
+        cls.country_2 = cls.env["res.country"].create({"name": "Test country 2"})
+        cls.state = cls.env["res.country.state"].create(
+            {"name": "Test state", "code": "TS", "country_id": cls.country_2.id}
+        )
+        cls.partner_2 = cls.env["res.partner"].create(
+            {
+                "name": "Test partner 2",
+                "country_id": cls.country_2.id,
+                "state_id": cls.state.id,
+                "zip": "22222",
+            }
+        )
+        cls.partner_3 = cls.env["res.partner"].create(
+            {
+                "name": "Test partner 3",
+                "country_id": cls.country_2.id,
+                "state_id": cls.state.id,
+                "zip": "33333",
+            }
+        )
+        cls.product = cls.env["product.product"].create(
+            {"name": "Test carrier multi", "detailed_type": "service"}
+        )
+        cls.product_child_1 = cls.env["product.product"].create(
+            {"name": "Test child 1", "detailed_type": "service"}
+        )
+        cls.product_child_2 = cls.env["product.product"].create(
+            {"name": "Test child 2", "detailed_type": "service"}
+        )
+        cls.carrier_multi = cls._create_carrier(
+            cls,
+            (
+                {
+                    "name": "Test child 1",
+                    "product_id": cls.product_child_1,
+                    "zip_from": 20000,
+                    "zip_to": 29999,
+                    "fixed_price": 50,
+                },
+                {
+                    "name": "Test child 2",
+                    "product_id": cls.product_child_2,
+                    "zip_from": 30000,
+                    "zip_to": 39999,
+                    "fixed_price": 150,
+                },
+            ),
+        )
+        cls.carrier_single = cls.carrier_multi.copy(
+            {
+                "name": "Test carrier single",
+                "destination_type": "one",
+                "child_ids": False,
+            }
+        )
+        cls.product = cls.env["product.product"].create(
+            {"name": "Test product", "detailed_type": "product", "list_price": 1}
+        )
+        cls.sale_order = cls._create_sale_order(cls)
+
+    def _create_carrier(self, childs):
+        carrier_form = Form(self.env["delivery.carrier"])
+        carrier_form.name = "Test carrier multi"
+        carrier_form.product_id = self.product
+        carrier_form.destination_type = "multi"
+        carrier_form.delivery_type = "fixed"
+        carrier_form.fixed_price = 100
+        for child_item in childs:
+            with carrier_form.child_ids.new() as child_form:
```

Comment:
Is this really needed to be created by `Form`? And why extract it as method if it's called only once?

Reply:
The idea of using `Form` is to check that UX process is correct. 
Extracting the method was to make it more readable than a very long setup. 
Do you prefer to leave it as it was before?

Reply:
In this case it's not adding too much value.